src/features/instance/applications/new/functions/isValidTarballUrl.ts

```diff
@@ -0,0 +1,11 @@
+import isValidUrl from '@/shared/functions/isValidUrl';
+
+function isValidTarballUrl(url: string) {
+	// npm restrictions on the tarball url install here: https://docs.npmjs.com/cli/v9/commands/npm-install
+	// updated to indexOf to allow signed URLs (with tokens appended)
+	return isValidUrl(url) && (url.indexOf('.tar') !== -1 || url.indexOf('.tar.gz') !== -1 || url.indexOf('.tgz') !== -1);
```

Comment:
I recommend using `includes()` for readability.

```typescript
return isValidUrl(url) && (url.includes('.tar') || url.includes('.tar.gz') || url.includes('.tgz'));
```

You could even do it all in one regex:

```typescript
return isValidUrl(url) && /\.(tar(\.gz)?|tgz)$/.test(url);
```